libcontainer/process_linux.go

```diff
@@ -836,6 +840,28 @@ func (p *initProcess) createNetworkInterfaces() error {
 	return nil
 }
 
+// setupNetworkDevices sets up and initializes any defined network interface inside the container.
+func (p *initProcess) setupNetworkDevices() error {
+	// host network pods does not move network devices.
+	if !p.config.Config.Namespaces.Contains(configs.NEWNET) {
+		return nil
+	}
+	// get the namespace defined by the config and fall back
+	// to the one created by runc to run the container process.
+	nsPath := p.config.Config.Namespaces.PathOf(configs.NEWNET)
+	if nsPath == "" {
+		nsPath = fmt.Sprintf("/proc/%d/ns/net", p.pid())
+	}
+	for name, netDevice := range p.config.Config.NetDevices {
+		err := devChangeNetNamespace(name, nsPath, *netDevice)
+		if err != nil {
+			return err
```

Comment:
I wonder what we should do if moving some ifaces work but some fail. Shall we move back to the init netns the ones that we moved to the container?

To be clear, this is an honest question, I don't know what would be the right thing to do. Maybe keep them in the new netns is the right thing, as the kernel will move it back whenever the netns is destroyed and we the code here is quite simple. But if we decide to do that, I'd put a comment here.

Reply:
I think we shouldn't try to restore them back nor anything, I'd let them fail and return the error as we do here. They will be restored by the kernel when the netns ceases to exist.

But I would add a comment here explaining that we don't handle it because of that.

Reply:
ack